upsetplot/data.py

```diff
@@ -16,3 +19,51 @@ def generate_data(seed=0, n_samples=10000, n_sets=3, aggregated=False):
     if aggregated:
         return df.value.groupby(level=list(range(n_sets))).count()
     return df.value
+
+
+def from_sets(data):
+    """
+    Data loader for a dict of sets
+
+    :param Dict[str, set[str]] data:
```

Comment:
This is underspecified. Consider items being lowercase and set names being uppercase. The following are both valid and equivalent representations of set intersections as dicts of sets:

```py
{'A': {'a', 'b', 'c', 'f'},
 'B': {'a', 'b', 'd'},
 'C': {'b', 'd', 'e', 'f', 'g'},
 }
```

```
{'a': {'A', 'B'},
 'b': {'A', 'B', 'C'},
 'c': {'A'},
 'd': {'B', 'C'},
 'e': {'C'},
 'f': {'A', 'C'},
 'g': {'C'},
 }
```

This is one reason I called my version `dict_of_collections`.

Indeed, the latter is more expressive as it is able to represent the case when an item is absent from all sets, while your format cannot. But it does not need to be a dict, so maybe we should separately support that input format as an iterable of `[(item, memberships), ...]` (or as parallel `(items, memberships)` which might work better where the items are in a dataframe and we want to plot distributions for set intersections, not just cardinality).